pages/docs/tutorials/native/working-with-klib.md

```diff
@@ -0,0 +1,140 @@
+---
+type: tutorial
+layout: tutorial
+title:  "Working with Kotlin/Native Libraries"
+description: "A look at how to work with Kotlin/Native libraries"
+authors: Hadi Hariri 
+date: 2018-01-22
+showAuthorInfo: false
+---
+
+
+In this tutorial we'll see
+
+* [What is a Kotlin/Native Libary](#what-is-kotlin/native-library)
+* [How to create a Kotlin/Native Library](#creating-a-kotlin/native-libraries)
+* [How to consume Kotlin/Native libraries](#consuming-a-kotlin/native-library)
+* [How to use and create library repositories](#using-and-creating-library-repositories)
+
+
+## What is Kotlin/Native Library
+
+The concept of a library, a collection of resources to be used by applications, is available in Kotlin/Native and it is represented
+as a file with the extension `klib`. Kotlin/Native libraries are artifacts that are only available at compile time, i.e. we do not ship these as dependencies to our application
+(as we would for instance with DLL files on the Windows platform).
+
+A `klib` file is a compressed archive in zip format with the following directory structure:
+
+File `utils.klib`
+
+```
+    targets/                    /* One or more target platforms */
+    
+        macbook/                /* When targeting macbook, this entry would appear */
+        
+            kotlin/             /* Kotlin code compiled to LLVM bitcode */
+            
+            native/             /* Bitcode for additonal native objects */ 
+            
+        linux/                  /* When targeting Linux, this entry would appear */
+        
+            kotlin/
+            
+            native/
+            
+    linkdata/                   /* A set of [ProtoBuf](https://github.com/google/protobuf) files with serialised linkage metadata */ 
+    
+    resources/                  /* Resource files such as images, etc. */
+
+    manifest                    /* A file in Java Property Format describing the library */ 
+```
+
+**Note**: It's important to note that until Kotlin/Native stabilises, the internal format of a `klib` file may change somewhat. Usually when working with `klib` files however,  
+knowing the internals is not required. 
```

Comment:
If it's not required to know, why is it the first thing you tell your readers about?

Reply:
Cause it's a warning and if someone comes across this tutorial and sees it's not this they might now why I've not updated it yet :). But point taken.